mkosi.conf.d/10-systemd.conf

```diff
@@ -13,6 +13,8 @@ OutputDirectory=mkosi.output
 [Content]
 BuildDirectory=mkosi.builddir
 Cache=mkosi.cache
+# Make gdb work properly by putting the source files in the same location as they were in during the build.
```

Comment:
there's the `-ffile-prefix-map=` build flag that can be passed if we want to override the path, if that makes it more sensible?

Reply:
You can do this at runtime with gdb `set substitute-path` so I added that to our .gdbinit file and moved the sources to /root/src instead.